src/settings.rs

```diff
@@ -27,22 +34,156 @@ impl<'src> Settings<'src> {
   }
 
   pub(crate) fn shell_binary<'a>(&'a self, config: &'a Config) -> &'a str {
-    if let (Some(shell), false) = (&self.shell, config.shell_present) {
+    let shell_or_args_present = config.shell.is_some() | config.shell_args.is_some();
```

Comment:
It's better to use `||` instead of `|`. They both work, but for clarity, `|` should only be used when you want bitwise or.

```suggestion
    let shell_or_args_present = config.shell.is_some() || config.shell_args.is_some();
```